packages/x-charts-pro/src/FunnelChart/useFunnelChartProps.ts

```diff
@@ -0,0 +1,150 @@
+'use client';
+import { DEFAULT_MARGINS, DEFAULT_X_AXIS_KEY, DEFAULT_Y_AXIS_KEY } from '@mui/x-charts/constants';
+import { ChartsOverlayProps } from '@mui/x-charts/ChartsOverlay';
+import { ChartsAxisProps } from '@mui/x-charts/ChartsAxis';
+import { ChartsLegendSlotExtension } from '@mui/x-charts/ChartsLegend';
+import useId from '@mui/utils/useId';
+import { ChartsClipPathProps } from '@mui/x-charts/ChartsClipPath';
+import { ChartsWrapperProps, defaultizeMargin } from '@mui/x-charts/internals';
+import { ChartsAxisHighlightProps } from '@mui/x-charts/ChartsAxisHighlight';
+import { FunnelPlotProps } from './FunnelPlot';
+import type { FunnelChartProps } from './FunnelChart';
+import { ChartContainerProProps } from '../ChartContainerPro';
+
+/**
+ * A helper function that extracts FunnelChartProps from the input props
+ * and returns an object with props for the children components of FunnelChart.
+ *
+ * @param props The input props for FunnelChart
+ * @returns An object with props for the children components of FunnelChart
+ */
+export const useFunnelChartProps = (props: FunnelChartProps) => {
+  const {
+    xAxis,
+    yAxis,
+    series,
+    width,
+    height,
+    margin: marginProps,
+    colors,
+    sx,
+    children,
+    slots,
+    slotProps,
+    skipAnimation,
+    loading,
+    onItemClick,
+    highlightedItem,
+    onHighlightChange,
+    className,
+    hideLegend,
+    axisHighlight,
+    apiRef,
+    ...rest
+  } = props;
+  const margin = defaultizeMargin(marginProps, DEFAULT_MARGINS);
+
+  const id = useId();
+  const clipPathId = `${id}-clip-path`;
+
+  const defaultBandAxisConfig = {
+    scaleType: 'band',
+    categoryGapRatio: 0,
+    data: Array.from(
+      { length: Math.max(...series.map((s) => (s.data ?? []).length)) },
+      (_, index) => index,
+    ),
+  } as const;
+
+  const defaultLinearAxisConfig = {
+    scaleType: 'linear',
+    domainLimit: 'strict',
+  } as const;
+
+  const isHorizontal = series.some((s) => s.layout === 'horizontal');
```

Comment:
If one series is vertical and another one is horizontal, what's the expected behavior? It seems a bit weird that the API allows configurations that are contradictory.

Reply:
Is there a card for that? We can move the discussion there. I wonder if there are some use case where it might make sense to have layout on a per-series level.

Reply:
It doesn't look like there is an issue for that, and I can't think of a use-case for it. Maybe @alexfauquette knows better.